framework/src/base/MooseObject.C

```diff
@@ -34,6 +34,7 @@ InputParameters
 validParams<MooseObject>()
 {
   InputParameters params;
+  params.addRequiredParam<std::string>("type", "The C++ type to instantiate.");
```

Comment:
This is a little wonky. The type is the type of object. I can see why you are adding this, but I don't think I necessarily like it. It's always redundant.